processing/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -370,6 +370,18 @@ public static ExprEval ofComplex(ExpressionType outputType, @Nullable Object val
     return new ComplexExprEval(outputType, value);
   }
 
+  public static ExprEval bestEffortArray(@Nullable List<?> theList)
+  {
+    // do not convert empty lists to arrays with a single null element here, because that should have been done
+    // by the selectors preparing their ObjectBindings if necessary. If we get to this point it was legitimately
+    // empty
+    NonnullPair<ExpressionType, Object[]> coerced = coerceListToArray(theList, false);
+    if (coerced == null) {
+      return bestEffortOf(null);
```

Comment:
I think it would probably be best to have null handled first in `bestEffortOf` so that if we ever decide to represent `null` as something more sensible (like introduce a null type), we won't be making a 'string' here, not to mention the saving of not running through a bunch of checks for a null value.

Reply:
Sure, I expected you to actually to create a method for `makeNullNode()` or something in `ExprEval` instead of newing it up directly here.  Just moving around the ordering of the statements is 🤷 .  Though, checking for `null` first is probably better than last.